### ISSUE TITLE
twisted.pair.test fails under Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8744]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8744|
|Type|enhancement|
|Created|2016-08-11 06:03:01Z|
|Branch|https://github.com/twisted/twisted/tree/8744-rodrigc-pair-py3|

If I run:

```
tox -e py35-tests twisted.pair.test
```

it fails under Python 3

<details><summary>Searchable metadata</summary>

```
trac-id__8744 8744
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8744_rodrigc_pair_py3 8744-rodrigc-pair-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__pair pair
keywords__None None
time__1470895381701257 1470895381701257
changetime__1470928013401775 1470928013401775
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/484

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_3"></a>|@acabhishek942 commented|
|-|-|

changes looks good. Please merge

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 095d01d0f3964179af94b5ba88df5fd463d6838d
```
#!CommitTicketReference repository="" revision="095d01d0f3964179af94b5ba88df5fd463d6838d"
Merge pull request #484 from twisted/8744-rodrigc-pair-py3

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8744

Port twisted.pair to Python 3
```
